src/main/webapp/ts/matches.ts

```diff
@@ -26,4 +22,90 @@ function createMatchElement(match: Match) {
     matchElement.appendChild(nameElement);
 
     return matchElement;
-}
\ No newline at end of file
+}
+
+function openForm() {
+    document.getElementById("matchForm")!.style.display = "block";
+}
+
+function closeForm() {
+    document.getElementById("matchForm")!.style.display = "none";
+} 
+
+function sendMatchRequest() {
+    closeForm();
+
+    const productArea = (<HTMLInputElement>document.getElementById('product_area')).value;
+
+  // Create the request to send to the server using the data we collected from
+  // the web form.
+  const matchRequest = new MatchRequest(productArea);
+
+  queryServer(matchRequest).then((matches) => {
+      displayNewMatchPopup(matches);
+  });
+}
+
+function displayNewMatchPopup(matches : Array<Match>) {
+    const newMatchContainer = <HTMLSelectElement>document.getElementById('new-matches');
+
+  // clear out any old results
+  newMatchContainer.innerHTML = '';
+
+  // add results to the page
+  for (const match of matches) {
+    const matchString : string = matchToString(match);
+    let newOption = new Option(matchString, matchString);
+    newMatchContainer.add(newOption, undefined);
+  }
+}
+
+
+function matchToString(match : Match) {
+    return match.name;
+}
+
+/**
+ * Sends the match request to the server and get back the matches.
+ */
+async function queryServer(matchRequest: MatchRequest) {
+  const json = JSON.stringify(matchRequest);
+  return fetch('/new-matches-query', {method: 'POST', body: json})
+      .then((response) => {
+        return response.json();
+      })
+      .then((users) => {
+        // Convert the range from a json representation to our Match class.
+        const out : Array<Match> = [];
+        users.forEach((range: User) => {
+          out.push(new Match(range.name));
+        });
+        return out;
+      });
+}
+
+class MatchRequest {
+    productArea: string;
+    constructor(productArea: string) {
+        this.productArea = productArea;
+    }
+}
+
+class User {
+    name: string;
+    constructor(name: string) {
+        this.name = name;
+    }
+
+    toMatch() {
+        let newMatch : Match = new Match(this.name);
+        return newMatch;
+    }
+}
+
+class Match {
+    name: string;
```

Comment:
name is fine for now, just a reminder that this ultimately will probably need to be some kind of ID (as previously mentioned, since names aren't unique / can be non ascii, etc)

Reply:
This is on my to-do list :)